server/src/main/java/pl/edu/agh/RaftServer.java

```diff
@@ -215,9 +246,12 @@ private void commitEntry(LogEntry entry) {
                     .toBlocking()
                     .first();
 
-            connection.getInput().forEach(byteBuf -> {
-                handleResponse(MessageUtils.toObject(byteBuf.toString(Charset.defaultCharset())));
-            });
+            connection.getInput()
+                    .subscribe(
+                            byteBuf -> handleResponse(MessageUtils.toObject(byteBuf.toString(Charset.defaultCharset()))),
```

Comment:
I'm not sure if this is the only place where we should add it. I think we could leave it for now.